Sources/WebSocketKit/WebSocket.swift

```diff
@@ -16,14 +16,19 @@ public final class WebSocket {
     }
 
     public private(set) var isClosed: Bool
+    public private(set) var closedCode: WebSocketErrorCode?
 
-    public var onClose: EventLoopFuture<Void> {
-        return self.channel.closeFuture
+    public var onClose: EventLoopFuture<WebSocketErrorCode?> {
```

Comment:
This is a breaking change. I think we should leave this as `Void` especially if you can access `self.closedCode` on the WebSocket.

Reply:
You're right, if the closed code stays (as above), will put this back.

Reply:
I think the reason I did this was that this is headed to a v2 correct?

Reply:
Yeah breaking changes are acceptable ATM, but still should have a good reason to make them.

Reply:
That's fair, will change back.